BatteryHealthChecker.c

```diff
@@ -0,0 +1,106 @@
+#include "BatteryHealthChecker.h"
+
+/*=============================================================================
+=======                                MACROS                           =======
+=============================================================================*/
+#define MAXTEMP 45.0F
+#define MINTEMP 0.0F
+#define MAXSOC 80.0F
+#define MINSOC 20.0F
+#define CHARGERATETHRESHOLD 0.8F
+
+/*********************************************************************************
+ * A function that checks the lower range of the passed attribute
+ * attribute can be temperature , state of charge or Charge Rate
+ * Thresholds has been defined globally
+ *********************************************************************************/
+int LowerThresholdChecker(float attribute, float lowerthreshold)
+{
+  bool isValid = attribute < lowerthreshold;
+  if(isValid)
+    return 0;
+  else return 1;
+}
+/*********************************************************************************
+ * A function that checks the higher range of the passed attribute
+ * attribute can be temperature , state of charge or Charge Rate
+ * Thresholds has been defined globally
+ *********************************************************************************/
+int UpperThresholdChecker(float attribute, float upperthreshold)
+{
+  bool isValid = attribute > upperthreshold;
+  if(isValid)
+    return 0;
+  else return 1;
+}
+
+/*********************************************************************************
+ * A function that prints the temperature abnormality
+ *********************************************************************************/
+void FaultyMessageTemperature(int TemperatureStatus)
```

Comment:
You have made separate checks for low and high, but the output doesn't differentiate. So a user/system cannot take action on it. Try to make that differentiation.